ocaml/xapi/repository_helpers.ml

```diff
@@ -817,6 +817,22 @@ let assert_url_is_valid ~url =
     error "Invalid url %s: %s" url (ExnHelper.string_of_exn e) ;
     raise Api_errors.(Server_error (invalid_base_url, [url]))
 
+let gpgkey_path_pattern = Re.Str.regexp "^[A-Za-z0-9_-]+$"
+
+let assert_gpgkey_path_is_valid path =
+  (* When Xapi_globs.repository_gpgcheck is:
+   * true, an empty gpgkey path will
+   *   1) reuslt to use default one which is configured in repository-gpgkey-path, or,
+   *   2) raise an error to user if no default one configured;
+   * false, an empty gpgkey path will be ignored.
+   * The existence and validaty of the GPG public key file will be verified before using *)
+  if path = "" || Re.Str.string_match gpgkey_path_pattern path 0 then
```

Comment:
Avoid the `Re.Str` module because its thread safety is not clear https://discuss.ocaml.org/t/re-str-is-not-thread-safe/9425/17 and use `Re.Posix`

Reply:
Re might be not thread-safe after all. I'd propose instead:

```ocaml
let gpgkey_path_is_valid = function
| 'A' .. 'Z' | 'a'..'z' | '0'..'9' | '_' | '-' -> true
| _ -> false

let assert_gpgkey_path_is_valid path =
  if path = "" || String.forall gpgkey_path_is_valid path then
```
Technicallthe empty string check could be removed and still be valid, but it's better to have it explicitly.